backend/ohq/models.py

```diff
@@ -447,3 +448,70 @@ class Meta:
 
     def __str__(self):
         return f"{self.user}: {self.metric}"
+    
+class Booking(models.Model):
+    """
+    Booking within an occurrence
+    """
+
+    occurrence = models.ForeignKey(Occurrence, on_delete=models.CASCADE, related_name="bookings")
+    user = models.ForeignKey(User, on_delete=models.CASCADE, blank=True, null=True)
+    start = models.DateTimeField("start", db_index=True)
+    end = models.DateTimeField("end", db_index=True)
+
+    class Meta:
+        verbose_name = ("booking")
+        verbose_name_plural = ("bookings")
+        ordering = ["start"]
+        index_together = (("start", "end"),)
+
+    def clean(self):
+        if self.start >= self.end:
+            raise ValidationError('Start time must be before end time.')
+        
+        if self.start.minute % 15 != 0:
+            raise ValidationError('Start time must be on a 15-minute interval (e.g., :00, :15, :30, :45).')
+        
```

Comment:
Can we document this behavior somewhere (either top level in code, in PR description, etc)

Also looping in @ryantanen who will likely be doing the frontend. If we are not allowing non-15-min intervals, we should also prevent booking non-15-min slots on the frontend as well.